opencog/haskell/TruthValue_CWrapper.cpp

```diff
@@ -0,0 +1,107 @@
+#include "TruthValue_CWrapper.h"
+
+
+int TruthValue_toRawType(TruthValuePtr tv
+                        , char** tv_type
+                        , double* parameters)
+{
+    const std::string & tvt = classserver().getTypeName(tv->getType());
+
+    *tv_type = (char*) malloc(sizeof(char) * (tvt.length()+1));
+    if(! *tv_type)
+        throw RuntimeException(TRACE_INFO,"Failed malloc.");
+    std::strcpy(*tv_type, tvt.c_str());
+
+    if (strcmp(*tv_type,"SimpleTruthValue") == 0){
```

Comment:
Also, you might want to think about eventually supporting other kinds of values: not just truth values, but generically, string values, and link values (which are just lists of other values ... i.e. the values are now fully hierarchical).

If this is unclear, its because I am writing the wiki documentation for this now.  See issue #513 for other status and details